### ISSUE TITLE
AWS custom runtime seems to ignore content type from API gateway

### ISSUE THREAD

**Issue by dsyer** [spring-cloud/spring-cloud-function#530]
I'm not 100% sure how this happens. What I'm trying to do is run a function in AWS with a custom runtime (so `HttpSupplier` and `FunctionExporter`) and trigger with an API Gateway. The API Gateway gets a POST with a `Content-Type`, but that doesn't seem to do anything - the conversion is always to a String.

**Comment by dsyer** [comment]
This is what I see inbound in CloudWatch (when I crank up the DEBUG logging):

```
2020-05-28T07:48:37.069+00:00
2020-05-28 07:48:37.069 DEBUG 12 --- [ctor-http-nio-1] o.s.c.f.c.c.SimpleFunctionRegistry       : Raw type of value: GenericMessage [payload=
{
    "resource": "/native",
    "path": "/native",
    "httpMethod": "POST",
    "headers": {
        "accept": "*/*",
        "content-type": "application/json",
        "Host": "rfk9r4kgg1.execute-api.eu-west-1.amazonaws.com",
        "User-Agent": "curl/7.68.0",
        "X-Amzn-Trace-Id": "Root=1-5ecf6cd5-8f8221e8f4ab44203a9a0fc8",
        "X-Forwarded-For": "86.30.187.113",
        "X-Forwarded-Port": "443",
        "X-Forwarded-Proto": "https"
    },
    "multiValueHeaders": {
        "accept": [
            "*/*"
        ],
        "content-type": [
            "application/json"
        ],
        "Host": [
            "rfk9r4kgg1.execute-api.eu-west-1.amazonaws.com"
        ],
        "User-Agent": [
            "curl/7.68.0"
        ],
        "X-Amzn-Trace-Id": [
            "Root=1-5ecf6cd5-8f8221e8f4ab44203a9a0fc8"
        ],
        "X-Forwarded-For": [
            "86.30.187.113"
        ],
        "X-Forwarded-Port": [
            "443"
        ],
        "X-Forwarded-Proto": [
            "https"
        ]
    },
    "queryStringParameters": null,
    "multiValueQueryStringParameters": null,
    "pathParameters": null,
    "stageVariables": null,
    "requestContext": {
        "resourceId": "x65mc5",
        "resourcePath": "/native",
        "httpMethod": "POST",
        "extendedRequestId": "NO3xTF4tDoEFSzg=",
        "requestTime": "28/May/2020:07:48:37 +0000",
        "path": "/default/native",
        "accountId": "816194980775",
        "protocol": "HTTP/1.1",
        "stage": "default",
        "domainPrefix": "rfk9r4kgg1",
        "requestTimeEpoch": 1590652117053,
        "requestId": "529a0f1e-93b6-42c1-93f4-cb028087de5d",
        "identity": {
            "cognitoIdentityPoolId": null,
            "accountId": null,
            "cognitoIdentityId": null,
            "caller": null,
            "sourceIp": "86.30.187.113",
            "principalOrgId": null,
            "accessKey": null,
            "cognitoAuthenticationType": null,
            "cognitoAuthenticationProvider": null,
            "userArn": null,
            "userAgent": "curl/7.68.0",
            "user": null
        },
        "domainName": "rfk9r4kgg1.execute-api.eu-west-1.amazonaws.com",
        "apiId": "rfk9r4kgg1"
    },
    "body": "{\"value\":\"Poo\"}",
    "isBase64Encoded": false
}
, headers={date=Thu, 28 May 2020 07:48:37 GMT, scf-sink-url=http://127.0.0.1:9001/2018-06-01/runtime/invocation/{{destination}}/response, scf-func-name=origin|foobar, lambda-runtime-trace-id=Root=1-5ecf6cd5-8f8221e8f4ab44203a9a0fc8;Parent=26cabad502e71de1;Sampled=0, lambda-runtime-aws-request-id=f291236e-0f51-43dd-9d4b-977d543714f1, id=5ceedeb2-6523-51d7-fc1a-4ac2be3c9bed, lambda-runtime-invoked-function-arn=arn:aws:lambda:eu-west-1:816194980775:function:native, lambda-runtime-deadline-ms=1590652120066, timestamp=1590652117069}]is java.util.Map<java.lang.String, java.lang.Object>
```

So the content type is there, but it seems to be embedded in the payload, which is no use at all.

**Comment by olegz** [comment]
Resolved with a1d10f07712980d65bb4d9da4caaf3c1bd6ec11b
So, now we get the content-type from the event response before constructing an input message that goes into function input

**Comment by olegz** [comment]
Here is an example input Message generated after API Gateway request
```

 Event message: GenericMessage [payload=byte[4], headers={date=Fri, 22 Jan 2021 10:04:12 GMT, content-length=1329, lambda-runtime-trace-id=Root=1-600aa31c-198b81aff209fcdfbda52b4f;Parent=1dcfd3417a3a35f5;Sampled=0, resource=/echocustom, lambda-runtime-aws-request-id=37ee6135-21b7-4361-9c7b-1f84ee3405b6, httpMethod=POST, lambda-runtime-invoked-function-arn=arn:aws:lambda:eu-west-3:313369169943:function:lambda-custom, path=/echocustom, isBase64Encoded=false, requestContext={resourceId=ewg3px, resourcePath=/echocustom, httpMethod=POST, extendedRequestId=Zi5seFV5CGYFr7g=, requestTime=22/Jan/2021:10:04:12 +0000, path=/echocustom, accountId=313369169943, protocol=HTTP/1.1, stage=test-invoke-stage, domainPrefix=testPrefix, requestTimeEpoch=1611309852539, requestId=03b0f20f-5f2c-4b91-be7e-92a54af56777, identity={. . .}, id=1e346a33-fd8a-332f-05a3-0402342a59e5, contentType=application/json, lambda-runtime-deadline-ms=1611309855557, timestamp=1611309852560}]

```
